pkg/kubelet/kuberuntime/logs/logs_test.go

```diff
@@ -454,3 +453,67 @@ func TestReadLogsLimitsWithTimestamps(t *testing.T) {
 
 	assert.Equal(t, 2, lineCount, "should have two lines")
 }
+
+func TestReadLogsWithLogRotation(t *testing.T) {
+	logLineFmt := "2023-06-09T05:28:57.806368292Z stdout F %v\n"
+
+	dir, err := os.MkdirTemp("", "readlogs")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dir)
+	tmpfile, err := os.CreateTemp(dir, "log.*.txt")
+	assert.NoError(t, err)
+
+	for i := 0; i < 5; i++ {
+		tmpfile.WriteString(fmt.Sprintf(logLineFmt, i))
+	}
+	tmpfile.Close()
+
+	ctx := context.Background()
+	// use fake runtime service
+	containerID := "83a21271e2a84a48ac89-6411e55fb8c1"
+	fakeRuntimeService := apitesting.NewFakeRuntimeService()
+	fakeRuntimeService.Containers[containerID] = &apitesting.FakeContainer{
+		ContainerStatus: runtimeapi.ContainerStatus{
+			Id:    containerID,
+			State: runtimeapi.ContainerState_CONTAINER_RUNNING,
+		},
+	}
+
+	// mock log rotation
+	go func() {
+		time.Sleep(1 * time.Second)
+
+		os.Rename(tmpfile.Name(), tmpfile.Name()+".1")
+		defer os.Remove(tmpfile.Name() + ".1")
+
+		// call CRI.ReopenContainerLog took some time
+		time.Sleep(500 * time.Millisecond)
+		newTmpfile, err := os.OpenFile(tmpfile.Name(), os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+		assert.NoError(t, err)
+
+		for i := 5; i < 10; i++ {
+			newTmpfile.WriteString(fmt.Sprintf(logLineFmt, i))
+		}
+		newTmpfile.Close()
```

Comment:
```
		newTmpfile, err := os.OpenFile(tmpfile.Name(), os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
		assert.NoError(t, err)

		for i := 5; i < 10; i++ {
			newTmpfile.WriteString(fmt.Sprintf(logLineFmt, i))
		}
		newTmpfile.Close()
```
This looks like a good candidate to be wrapped into function and called here and when the file was created first time.

Reply:
Done.